submit/main/notify.py

```diff
@@ -59,33 +63,57 @@ def prepare_upload(file: FileStorage):
 
 
 def send_confirmation_emails(
-    excel_file: FileStorage, fund: str, reporting_period: str, fund_email: str, user_email: str, metadata: dict
-):
-    """Sends a confirmation email to the LA that submitted and to the Fund Team (with a link to the submission).
+    excel_file: FileStorage,
+    fund: FundConfig,
+    user_email: str,
+    metadata: dict,
+) -> None:
+    """Sends a confirmation email to the LA that submitted and to the Fund Team
+    with a link to the submission.
 
     :param excel_file: Excel file that has been submitted successfully
-    :param fund: the fund name for the submission
-    :param reporting_period: the reporting period for the submission
-    :param fund_email: the fund's email address to send confirmation to
+    :param fund: FundConfig object for the fund that the submission was made to
     :param user_email: the user's email address to send confirmation to
     :param metadata: contains information about the submission
+
     :return: None
     """
-    personalisation = get_personalisation(excel_file, fund, reporting_period, metadata)
+
+    personalisation = get_personalisation(excel_file, fund, metadata)
+
     current_app.logger.info("Sending confirmation emails to LA and Fund Team")
-    # to the Local Authority
-    send_email(email_address=user_email, template_id=Config.LA_CONFIRMATION_EMAIL_TEMPLATE_ID, **personalisation)
-    # to Fund - includes the file
+
+    # email Local Authority
     send_email(
-        email_address=fund_email,
+        email_address=user_email,
+        template_id=Config.LA_CONFIRMATION_EMAIL_TEMPLATE_ID,
+        **personalisation,
+    )
+
+    programme_id = metadata.get("Programme ID")
+    round_number = fund.current_reporting_round
+
+    programme = get_programme_by_id_and_round(programme_id, round_number)
+    _, submission_id = get_or_generate_submission_id(programme, round_number, fund.fund_code)
+
+    object_name = f"{fund.fund_code}/{submission_id}"
+    personalisation.update(
+        {
+            "download_url": f"{Config.FIND_SERVICE_BASE_URL}/retrieve-spreadsheet/{object_name}",
+        }
+    )
```

Comment:
If all of this is just to generate the bit of personalisation, should we move it to `get_personalisation`? Also not sure I love this needing to make DB calls from here - can we just pass in the submission_id to `send_confirmation_emails` if that's what we need here?

I particularly think it's awkward to be calling `get_or_generate_submission_id` - that feels very much like a thing that should remain internal to `data_store`. Especially because it can just 'imagine' submission IDs (ie generate one if it can't find a match - but it wouldn't persist it anywhere)

Reply:
I got rid of the `get_personalisation` function. You'll let me know what you thing of the new version.

Reply:
Resolving this.